.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,83 @@
+name: guillotina
+
+on: [push]
+
+jobs:
+
+  # Job to run pre-checks
+  pre-checks:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8]
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install package
+        run: |
+          pip install $(grep "Cython" requirements.txt)
+          pip install -r requirements.txt
+          pip install -r contrib-requirements.txt
+          pip install -e .[test]
+          pip install -e .[testdata]
+      - name: Run pre-checks
+        run: |
+          flake8 guillotina --config=setup.cfg
+          mypy guillotina/ --ignore-missing-imports
+          isort -c -rc guillotina/
+          black --check --verbose guillotina
+  # Job to run tests
+  tests:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8]
+        database: ["DUMMY", "postgres", "cockroachdb"]
+        db_schema: ["custom", "public"]
+
+    # Set environment variables
+    env:
+      DATABASE: ${{ matrix.database }}
+      DB_SCHEMA: ${{ matrix.db_schema }}
+
+    steps:
+      - name: Check supported matrix combinations
+        if: (env.DATABASE == 'postgres' || env.DB_SCHEMA == 'public') != true
+        run: |
+          echo "This matrix combination is skipped"
+      - name: Checkout the repository
+        if: env.DATABASE == 'postgres' || env.DB_SCHEMA == 'public'
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        if: env.DATABASE == 'postgres' || env.DB_SCHEMA == 'public'
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install the package
```

Comment:
Why we need to reinstall ?

Reply:
because they run on different jobs.

I could also put all in one job if you like

Reply:
Maybe we could avoid installing everything on the pre-checks?